<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>Test_Prj1</class>
 <widget class="QWidget" name="Test_Prj1">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>1259</width>
    <height>663</height>
   </rect>
  </property>
  <property name="windowTitle">
   <string>程序烧写控制器</string>
  </property>
  <widget class="QPushButton" name="pB_exit">
   <property name="enabled">
    <bool>true</bool>
   </property>
   <property name="geometry">
    <rect>
     <x>10</x>
     <y>630</y>
     <width>75</width>
     <height>23</height>
    </rect>
   </property>
   <property name="cursor">
    <cursorShape>ArrowCursor</cursorShape>
   </property>
   <property name="text">
    <string>退出</string>
   </property>
  </widget>
  <widget class="QTextEdit" name="tE_log">
   <property name="geometry">
    <rect>
     <x>220</x>
     <y>20</y>
     <width>321</width>
     <height>591</height>
    </rect>
   </property>
   <property name="frameShape">
    <enum>QFrame::Box</enum>
   </property>
   <property name="lineWidth">
    <number>1</number>
   </property>
   <property name="autoFormatting">
    <set>QTextEdit::AutoNone</set>
   </property>
  </widget>
  <widget class="QPushButton" name="pB_clear">
   <property name="geometry">
    <rect>
     <x>110</x>
     <y>630</y>
     <width>75</width>
     <height>23</height>
    </rect>
   </property>
   <property name="text">
    <string>清空日志</string>
   </property>
  </widget>
  <widget class="QGroupBox" name="gB_udpserver">
   <property name="geometry">
    <rect>
     <x>10</x>
     <y>260</y>
     <width>191</width>
     <height>131</height>
    </rect>
   </property>
   <property name="title">
    <string>UDP服务器</string>
   </property>
   <widget class="QLabel" name="tL_port">
    <property name="geometry">
     <rect>
      <x>10</x>
      <y>60</y>
      <width>31</width>
      <height>16</height>
     </rect>
    </property>
    <property name="text">
     <string>port</string>
    </property>
   </widget>
   <widget class="QLineEdit" name="lE_port">
    <property name="geometry">
     <rect>
      <x>50</x>
      <y>60</y>
      <width>131</width>
      <height>20</height>
     </rect>
    </property>
    <property name="text">
     <string>9999</string>
    </property>
   </widget>
   <widget class="QComboBox" name="cB_ip">
    <property name="geometry">
     <rect>
      <x>50</x>
      <y>20</y>
      <width>131</width>
      <height>22</height>
     </rect>
    </property>
    <property name="editable">
     <bool>false</bool>
    </property>
    <property name="currentText">
     <string/>
    </property>
    <property name="currentIndex">
     <number>-1</number>
    </property>
   </widget>
   <widget class="QPushButton" name="pB_udpbegin">
    <property name="geometry">
     <rect>
      <x>90</x>
      <y>100</y>
      <width>75</width>
      <height>23</height>
     </rect>
    </property>
    <property name="text">
     <string>udp启动</string>
    </property>
   </widget>
   <widget class="QLabel" name="tL_ip">
    <property name="geometry">
     <rect>
      <x>10</x>
      <y>20</y>
      <width>21</width>
      <height>16</height>
     </rect>
    </property>
    <property name="text">
     <string>ip</string>
    </property>
   </widget>
  </widget>
  <widget class="QGroupBox" name="gB_udpclient">
   <property name="geometry">
    <rect>
     <x>10</x>
     <y>10</y>
     <width>191</width>
     <height>241</height>
    </rect>
   </property>
   <property name="title">
    <string>UDP客户端</string>
   </property>
   <widget class="QComboBox" name="cB_ip_2">
    <property name="enabled">
     <bool>true</bool>
    </property>
    <property name="geometry">
     <rect>
      <x>50</x>
      <y>20</y>
      <width>131</width>
      <height>22</height>
     </rect>
    </property>
    <property name="editable">
     <bool>true</bool>
    </property>
    <property name="currentText">
     <string/>
    </property>
    <property name="currentIndex">
     <number>-1</number>
    </property>
    <property name="duplicatesEnabled">
     <bool>false</bool>
    </property>
   </widget>
   <widget class="QLineEdit" name="lE_port_2">
    <property name="geometry">
     <rect>
      <x>50</x>
      <y>60</y>
      <width>131</width>
      <height>20</height>
     </rect>
    </property>
    <property name="text">
     <string>9999</string>
    </property>
   </widget>
   <widget class="QLabel" name="tL_ip_2">
    <property name="geometry">
     <rect>
      <x>10</x>
      <y>12</y>
      <width>21</width>
      <height>20</height>
     </rect>
    </property>
    <property name="text">
     <string>ip</string>
    </property>
   </widget>
   <widget class="QLabel" name="tL_port_2">
    <property name="geometry">
     <rect>
      <x>10</x>
      <y>60</y>
      <width>31</width>
      <height>16</height>
     </rect>
    </property>
    <property name="text">
     <string>port</string>
    </property>
   </widget>
   <widget class="QRadioButton" name="rB_byte">
    <property name="geometry">
     <rect>
      <x>100</x>
      <y>90</y>
      <width>89</width>
      <height>16</height>
     </rect>
    </property>
    <property name="text">
     <string>16进制</string>
    </property>
   </widget>
   <widget class="QRadioButton" name="rB_string">
    <property name="geometry">
     <rect>
      <x>10</x>
      <y>90</y>
      <width>91</width>
      <height>16</height>
     </rect>
    </property>
    <property name="layoutDirection">
     <enum>Qt::LeftToRight</enum>
    </property>
    <property name="text">
     <string>字符串</string>
    </property>
    <property name="checked">
     <bool>true</bool>
    </property>
   </widget>
   <widget class="QComboBox" name="cB_order">
    <property name="geometry">
     <rect>
      <x>20</x>
      <y>180</y>
      <width>151</width>
      <height>22</height>
     </rect>
    </property>
    <property name="editable">
     <bool>true</bool>
    </property>
    <property name="currentText">
     <string/>
    </property>
    <property name="currentIndex">
     <number>-1</number>
    </property>
   </widget>
   <widget class="QTextEdit" name="tE_send">
    <property name="geometry">
     <rect>
      <x>20</x>
      <y>120</y>
      <width>151</width>
      <height>51</height>
     </rect>
    </property>
    <property name="autoFormatting">
     <set>QTextEdit::AutoNone</set>
    </property>
   </widget>
   <widget class="QPushButton" name="pB_send">
    <property name="geometry">
     <rect>
      <x>90</x>
      <y>210</y>
      <width>71</width>
      <height>23</height>
     </rect>
    </property>
    <property name="text">
     <string>发送</string>
    </property>
   </widget>
  </widget>
  <widget class="QLineEdit" name="lE_incnt">
   <property name="geometry">
    <rect>
     <x>320</x>
     <y>510</y>
     <width>42</width>
     <height>20</height>
    </rect>
   </property>
   <property name="text">
    <string>9999</string>
   </property>
  </widget>
  <widget class="QLabel" name="tL_incnt">
   <property name="geometry">
    <rect>
     <x>235</x>
     <y>513</y>
     <width>71</width>
     <height>16</height>
    </rect>
   </property>
   <property name="text">
    <string>入库计数器：</string>
   </property>
  </widget>
  <widget class="QLabel" name="tL_outcnt">
   <property name="geometry">
    <rect>
     <x>235</x>
     <y>543</y>
     <width>71</width>
     <height>16</height>
    </rect>
   </property>
   <property name="text">
    <string>出库计数器：</string>
   </property>
  </widget>
  <widget class="QLineEdit" name="lE_outcnt">
   <property name="geometry">
    <rect>
     <x>320</x>
     <y>540</y>
     <width>41</width>
     <height>20</height>
    </rect>
   </property>
   <property name="text">
    <string>9999</string>
   </property>
  </widget>
  <widget class="QTextBrowser" name="tB_log">
   <property name="geometry">
    <rect>
     <x>210</x>
     <y>20</y>
     <width>391</width>
     <height>631</height>
    </rect>
   </property>
   <property name="maximumSize">
    <size>
     <width>16777215</width>
     <height>1000</height>
    </size>
   </property>
   <property name="mouseTracking">
    <bool>false</bool>
   </property>
   <property name="horizontalScrollBarPolicy">
    <enum>Qt::ScrollBarAsNeeded</enum>
   </property>
   <property name="sizeAdjustPolicy">
    <enum>QAbstractScrollArea::AdjustIgnored</enum>
   </property>
   <property name="autoFormatting">
    <set>QTextEdit::AutoAll</set>
   </property>
   <property name="tabChangesFocus">
    <bool>false</bool>
   </property>
   <property name="lineWrapMode">
    <enum>QTextEdit::WidgetWidth</enum>
   </property>
   <property name="acceptRichText">
    <bool>false</bool>
   </property>
   <property name="textInteractionFlags">
    <set>Qt::LinksAccessibleByKeyboard|Qt::LinksAccessibleByMouse|Qt::TextBrowserInteraction|Qt::TextEditable|Qt::TextSelectableByMouse</set>
   </property>
  </widget>
  <widget class="QTextBrowser" name="tB_log_2">
   <property name="geometry">
    <rect>
     <x>610</x>
     <y>20</y>
     <width>641</width>
     <height>631</height>
    </rect>
   </property>
   <property name="maximumSize">
    <size>
     <width>16777215</width>
     <height>1000</height>
    </size>
   </property>
   <property name="mouseTracking">
    <bool>false</bool>
   </property>
   <property name="horizontalScrollBarPolicy">
    <enum>Qt::ScrollBarAsNeeded</enum>
   </property>
   <property name="sizeAdjustPolicy">
    <enum>QAbstractScrollArea::AdjustIgnored</enum>
   </property>
   <property name="autoFormatting">
    <set>QTextEdit::AutoAll</set>
   </property>
   <property name="tabChangesFocus">
    <bool>false</bool>
   </property>
   <property name="lineWrapMode">
    <enum>QTextEdit::WidgetWidth</enum>
   </property>
   <property name="acceptRichText">
    <bool>false</bool>
   </property>
   <property name="textInteractionFlags">
    <set>Qt::LinksAccessibleByKeyboard|Qt::LinksAccessibleByMouse|Qt::TextBrowserInteraction|Qt::TextEditable|Qt::TextSelectableByMouse</set>
   </property>
  </widget>
  <widget class="QLabel" name="tL_txmac">
   <property name="geometry">
    <rect>
     <x>20</x>
     <y>390</y>
     <width>72</width>
     <height>15</height>
    </rect>
   </property>
   <property name="text">
    <string>发送MAC</string>
   </property>
  </widget>
  <widget class="QComboBox" name="cB_txmac">
   <property name="geometry">
    <rect>
     <x>20</x>
     <y>410</y>
     <width>161</width>
     <height>22</height>
    </rect>
   </property>
  </widget>
  <widget class="QPushButton" name="pB_udpbegin_2">
   <property name="geometry">
    <rect>
     <x>290</x>
     <y>160</y>
     <width>75</width>
     <height>23</height>
    </rect>
   </property>
   <property name="text">
    <string>udp设置</string>
   </property>
  </widget>
  <widget class="QCheckBox" name="chB_allip">
   <property name="geometry">
    <rect>
     <x>280</x>
     <y>100</y>
     <width>71</width>
     <height>16</height>
    </rect>
   </property>
   <property name="text">
    <string>全部</string>
   </property>
  </widget>
  <widget class="QGroupBox" name="groupBox">
   <property name="geometry">
    <rect>
     <x>10</x>
     <y>440</y>
     <width>181</width>
     <height>171</height>
    </rect>
   </property>
   <property name="title">
    <string>UWB距离精度</string>
   </property>
   <widget class="QPushButton" name="pB_avetime">
    <property name="geometry">
     <rect>
      <x>60</x>
      <y>130</y>
      <width>93</width>
      <height>28</height>
     </rect>
    </property>
    <property name="text">
     <string>开始统计</string>
    </property>
   </widget>
   <widget class="QLineEdit" name="lE_avetime">
    <property name="geometry">
     <rect>
      <x>70</x>
      <y>60</y>
      <width>81</width>
      <height>21</height>
     </rect>
    </property>
   </widget>
   <widget class="QLabel" name="tL_avetime">
    <property name="geometry">
     <rect>
      <x>20</x>
      <y>60</y>
      <width>41</width>
      <height>16</height>
     </rect>
    </property>
    <property name="text">
     <string>均值</string>
    </property>
   </widget>
   <widget class="QLabel" name="tL_avetime_2">
    <property name="geometry">
     <rect>
      <x>10</x>
      <y>100</y>
      <width>51</width>
      <height>16</height>
     </rect>
    </property>
    <property name="text">
     <string>标准差</string>
    </property>
   </widget>
   <widget class="QLineEdit" name="lE_avetime_2">
    <property name="geometry">
     <rect>
      <x>70</x>
      <y>100</y>
      <width>81</width>
      <height>21</height>
     </rect>
    </property>
   </widget>
   <widget class="QLabel" name="tL_avenum">
    <property name="geometry">
     <rect>
      <x>10</x>
      <y>20</y>
      <width>51</width>
      <height>20</height>
     </rect>
    </property>
    <property name="text">
     <string>数据量</string>
    </property>
   </widget>
   <widget class="QLineEdit" name="lE_avenum">
    <property name="geometry">
     <rect>
      <x>70</x>
      <y>20</y>
      <width>81</width>
      <height>21</height>
     </rect>
    </property>
   </widget>
  </widget>
  <zorder>gB_udpserver</zorder>
  <zorder>pB_exit</zorder>
  <zorder>tE_log</zorder>
  <zorder>pB_clear</zorder>
  <zorder>gB_udpclient</zorder>
  <zorder>lE_incnt</zorder>
  <zorder>tL_incnt</zorder>
  <zorder>tL_outcnt</zorder>
  <zorder>lE_outcnt</zorder>
  <zorder>tB_log_2</zorder>
  <zorder>tL_txmac</zorder>
  <zorder>cB_txmac</zorder>
  <zorder>pB_udpbegin_2</zorder>
  <zorder>chB_allip</zorder>
  <zorder>tB_log</zorder>
  <zorder>groupBox</zorder>
 </widget>
 <tabstops>
  <tabstop>pB_exit</tabstop>
 </tabstops>
 <resources/>
 <connections>
  <connection>
   <sender>pB_exit</sender>
   <signal>clicked()</signal>
   <receiver>Test_Prj1</receiver>
   <slot>close()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>37</x>
     <y>601</y>
    </hint>
    <hint type="destinationlabel">
     <x>199</x>
     <y>149</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>pB_clear</sender>
   <signal>clicked()</signal>
   <receiver>Test_Prj1</receiver>
   <slot>clearLog()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>110</x>
     <y>602</y>
    </hint>
    <hint type="destinationlabel">
     <x>550</x>
     <y>40</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>cB_ip_2</sender>
   <signal>editTextChanged(QString)</signal>
   <receiver>Test_Prj1</receiver>
   <slot>autoComp()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>107</x>
     <y>31</y>
    </hint>
    <hint type="destinationlabel">
     <x>110</x>
     <y>277</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>pB_udpbegin_2</sender>
   <signal>clicked()</signal>
   <receiver>Test_Prj1</receiver>
   <slot>udpSet()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>129</x>
     <y>112</y>
    </hint>
    <hint type="destinationlabel">
     <x>141</x>
     <y>290</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>pB_udpbegin</sender>
   <signal>clicked()</signal>
   <receiver>Test_Prj1</receiver>
   <slot>udpBegin()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>115</x>
     <y>447</y>
    </hint>
    <hint type="destinationlabel">
     <x>119</x>
     <y>133</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>tE_log</sender>
   <signal>textChanged()</signal>
   <receiver>Test_Prj1</receiver>
   <slot>cntUpdated()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>358</x>
     <y>264</y>
    </hint>
    <hint type="destinationlabel">
     <x>345</x>
     <y>312</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>pB_send</sender>
   <signal>clicked()</signal>
   <receiver>Test_Prj1</receiver>
   <slot>udpSend()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>160</x>
     <y>309</y>
    </hint>
    <hint type="destinationlabel">
     <x>161</x>
     <y>323</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>cB_order</sender>
   <signal>activated(QString)</signal>
   <receiver>Test_Prj1</receiver>
   <slot>udpOrder()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>64</x>
     <y>310</y>
    </hint>
    <hint type="destinationlabel">
     <x>66</x>
     <y>328</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>pB_avetime</sender>
   <signal>clicked()</signal>
   <receiver>Test_Prj1</receiver>
   <slot>uwbAveTime()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>84</x>
     <y>534</y>
    </hint>
    <hint type="destinationlabel">
     <x>82</x>
     <y>554</y>
    </hint>
   </hints>
  </connection>
 </connections>
 <slots>
  <slot>uwbBurn()</slot>
  <slot>bleBurn()</slot>
  <slot>setOK()</slot>
  <slot>udpBegin()</slot>
  <slot>onActivated()</slot>
  <slot>printLog()</slot>
  <slot>clearLog()</slot>
  <slot>sendString()</slot>
  <slot>sendByte()</slot>
  <slot>autoComp()</slot>
  <slot>udpSet()</slot>
  <slot>bleIn()</slot>
  <slot>bleOut()</slot>
  <slot>cntUpdated()</slot>
  <slot>udpSend()</slot>
  <slot>udpOrder()</slot>
  <slot>uwbAveTime()</slot>
 </slots>
</ui>
